elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreService.java

```diff
@@ -206,35 +227,89 @@ private AtomicLoadKeyStore.LoadKey load(AtomicLoadKeyStore keyStore) throws Exce
+                            break;
+                        case EXPIRED:
+                            ROOT_LOGGER.certificateExpired(keyStoreName, alias);
+                            break;
+                        case NOT_YET:
+                            ROOT_LOGGER.certificateNotYetValid(keyStoreName, alias);
+                            break;
+                        case VALID:
+                        default:
+                            break;
                     }
                 }
             }
         }
     }
 
+    private void scheduleCertificateHealthCheck() {
+        //JIC
+        cancelHealthCheck(false);
+        //Check if its one-time or periodic.
+        //Schedule in executor, to have consistent log output(prefix).
+        // simple fixed rate
+        final Runnable task =() -> {
+            try {
+                checkCertificatesValidity(keyStore);
+            } catch (Throwable e) {
+                e.printStackTrace();
+            }
+        };
+
+        if(this.expirationCheckDelay == 0) {
+            this.certificateTaskFuture = scheduledExecutorService.schedule(task, 10, TimeUnit.SECONDS);
+        } else {
+            this.certificateTaskFuture = scheduledExecutorService.scheduleAtFixedRate(task, 0, this.expirationCheckDelay, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private void cancelHealthCheck(final boolean interrupt) {
+        final ScheduledFuture<?> certificateTaskFuture = this.certificateTaskFuture;
+        this.certificateTaskFuture = null;
+        if (certificateTaskFuture != null) {
+            certificateTaskFuture.cancel(false);
+        }
+    }
+
+    static ScheduledExecutorService createScannerExecutorService() {
+        final ThreadFactory threadFactory = doPrivileged(new PrivilegedAction<ThreadFactory>() {
+            public ThreadFactory run() {
+                return new JBossThreadFactory(new ThreadGroup("ElytronKeyStore-threads"), Boolean.FALSE, null, "%G - %t", null, null);
+            }
+        });
+        return Executors.newScheduledThreadPool(1, threadFactory);
+    }
+
     @Override
     public void stop(StopContext stopContext) {
         ROOT_LOGGER.tracef(
                 "stopping:  keyStore = %s  unmodifiableKeyStore = %s  trackingKeyStore = %s  pathResolver = %s",
                 keyStore, unmodifiableKeyStore, trackingKeyStore, pathResolver
         );
-        keyStore = null;
-        unmodifiableKeyStore = null;
-        trackingKeyStore = null;
-        if (pathResolver != null) {
-            pathResolver.clear();
-            pathResolver = null;
+        try {
+            cancelHealthCheck(true);
+            keyStore = null;
+            unmodifiableKeyStore = null;
+            trackingKeyStore = null;
+            if (pathResolver != null) {
+                pathResolver.clear();
+                pathResolver = null;
+            }
+        } finally {
+            this.scheduledExecutorService.shutdownNow();
```

Comment:
Looking at how this is handled for the executors elsewhere a more complete shutdown solution is needed but the shared executor I am adding will have that,